public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -1339,16 +1340,29 @@ export class DashboardModel implements TimeModel {
     return this.originalDashboard;
   }
 
-  hasAngularPlugins(): boolean {
-    return this.panels.some((panel) => {
+  async hasAngularPlugins(): Promise<boolean> {
+    for (const panel of this.panels) {
+      let plugin;
+      if (!panel.plugin) {
+        try {
+          plugin = await dispatch(loadPanelPlugin(panel.type));
+          await panel.pluginLoaded(plugin);
```

Comment:
this logic as is written is kind of blocking, waiting in this line for each of the panels in the loop, is this your intention?

an alternative way of doing this without blocking waiting for each plugin to individually load but instead let the js engine handle the concurrency (if available):

```typescript
async hasAngularPlugins(): Promise<boolean> {
  await Promise.all( this.panels.map ( (panel) => {
     if (panel.plugin) { return } // Promise.all handle non-promises fine
     try {
        const  plugin = await dispatch(loadPanelPlugin(panel.type));
        //this is the promise we await for
        return panel.pluginLoaded(plugin);
     } catch(e) { return }
  });
  // ...
```

Reply:
Good catch! It's not necessary to wait for each plugin in the loop, I changed it to `Promise.all` :+1: